### PR TITLE
go.mod: Exclude prometheus/client_golang v1.12.1

### PR DESCRIPTION
**Pull request by mrueg**
See also https://github.com/prometheus/client_golang/issues/1012
As suggested in https://github.com/prometheus/client_golang/issues/1012#issuecomment-1090482644

### CODE REVIEW

**Review by kakkoyun — COMMENTED** [review]
LGTM. Thanks for taking care of this.

I guess Go 1.15 tests are failing, if we don't have a solution for it, I'm ok to drop the support for 1.15 since 1.18 is out.
I think having support for the last 3 major versions of Go is acceptable.

Better to do that in a separate PR though.

**Comment by kakkoyun** [comment]
Hey @mrueg, we plan to release soonish? Would you be able to have a look at this so that we can include it in the release?

**Comment by mrueg** [comment]
> Hey @mrueg, we plan to release soonish? Would you be able to have a look at this so that we can include it in the release?

Dropping go 1.15 support? Definitely can look into it.

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,7 @@ require (
 	github.com/beorn7/perks v1.0.1
 	github.com/cespare/xxhash/v2 v2.1.2
 	github.com/golang/protobuf v1.5.2
+	github.com/jpillora/backoff v1.0.0 // indirect
```
Comment (bwplotka) [inline go.mod:7]: This looks off. Do you mind doing `go mod tidy`?

Comment (mrueg) [reply go.mod:7]: yeah that got added with `go mod tidy`

```
go mod why github.com/jpillora/backoff
# github.com/jpillora/backoff
github.com/prometheus/client_golang/api/prometheus/v1
github.com/prometheus/client_golang/api/prometheus/v1.test
github.com/prometheus/common/config
github.com/mwitkow/go-conntrack
github.com/jpillora/backoff
```

**Review by bwplotka — APPROVED** [review]
Thanks! 💪🏽

**Pull request merged**